dvc/stage.py

```diff
@@ -493,18 +496,7 @@ def create(
         stage.wdir = wdir
         stage.path = path
 
-        # NOTE: remove outs before we check build cache
```

Comment:
Why did we move this outside?

Reply:
The change you are working on should work for `import-url` too.

Reply:
?

Reply:
> Why did we move this outside all the way up to dvc/repo/run.py?

I'm deleting this because it implies an old / deprecated use of `--remove-outs`.

> The change you are working on should work for import-url too.

Yep, will consider it in the next commit :+1: